lib/azure/role.rb

```diff
@@ -378,21 +396,22 @@ def setup(params)
                 end
 
                 if params[:tcp_endpoints]
-                  params[:tcp_endpoints].split(',').each do |endpoint|
-                    ports = endpoint.split(':')
-                    if !(ports.length > 1 && ports[1] == params[:port] || ports.length == 1 && ports[0] == params[:port])
+                  params[:tcp_endpoints].split(',').map(&:strip).each do |endpoint|
+                    ports = endpoint.split(':').map(&:strip)
+                    index = ports.length > 1 ? 1 : 0
+                    if !(ports[index] == params[:port])
                       xml.InputEndpoint {
                         xml.LocalPort ports[0]
-                        xml.Name 'tcpport_' + ports[0] + '_' + params[:azure_vm_name]
-                        if ports.length > 1
-                          xml.Port ports[1]
+                        if TCP_ENDPOINTS_MAPPING.keys.include?(ports[index])
+                          xml.Name TCP_ENDPOINTS_MAPPING[ports[index]]
                         else
-                          xml.Port ports[0]
+                          xml.Name "tcpport_#{ports[0]}_#{params[:azure_vm_name]}"
```

Comment:
Why do we need to add the vm name to the end? Everything works fine without it, right? It might be more useful to add something to it like "_chef' so that if someone looks at the endpoint they'll know it was provisioned by running Chef tools.